src/main/java/org/somerville/swag/data/entity/state/Guest.java

```diff
@@ -52,18 +87,47 @@ public void viewBasket() {
     }
 
     @Override
-    public void addProductToBasket() {
+    public void addProductToBasket(Product product, int quantity) {
+        customer.getCurrentOrder().getOrderLines().removeIf(orderLine -> orderLine.getProduct().getProductId() == product.getProductId());
+        customer.getCurrentOrder().add(new OrderLine(product, quantity));
+        product.setStockLevel(product.getStockLevel() - quantity);
 
+        // TODO: Write new Product Stock Level To Database
     }
 
     @Override
-    public void removeProductFromBasket() {
+    public void removeProductFromBasket(OrderLine orderLine) {
+        customer.getCurrentOrder().getOrderLines().remove(orderLine);
+        Product selectedProduct = orderLine.getProduct();
+        int quantity = orderLine.getQuantity();
+        selectedProduct.setStockLevel(selectedProduct.getStockLevel() - quantity);
 
+        // TODO: Write New Product Stock Level to Database
```

Comment:
Was our design decision to only update product stock level after checkout? so removing item from basket shouldn't require DB?